### PR TITLE
Add option to remove from session properties

### PR DESCRIPTION
**Pull request by basarhamdionat**
Added a new strategy to enable an option to remove certain fields from session properties when running verifier test

Test plan - (Please fill in how you tested your changes)
Added & ran unit test
== NO RELEASE NOTE ==

### CODE REVIEW

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfiguration.java [hunk 1]
```diff
@@ -57,10 +59,18 @@ public QueryConfiguration applyOverrides(QueryConfigurationOverrides overrides)
...
+                for (Entry<String, String> entry : overrides.getSessionPropertiesOverride().entrySet()) {
+                    sessionProperties.put(entry.getKey(), entry.getValue());
+                }
+            }
+            if (overrides.getSessionPropertiesOverrideStrategy() == REMOVE || overrides.getSessionPropertiesOverrideStrategy() == SUBSTITUTE_AND_REMOVE) {
+                sessionProperties = new HashMap<>(sessionProperties);
+                for (String sessionPropertyToRemove : overrides.getSessionPropertiesRemoval()) {
+                    sessionProperties.remove(sessionPropertyToRemove);
+                }
             }
```
Comment (mayankgarg1990) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfiguration.java:74]: I don't think that we need 2 separate settings for this - `SUBSTITUTE_AND_REMOVE` and `REMOVE`. If someone sets something to be removed - lets just remove them irrespective of the approach.

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfigurationOverrides.java [hunk 2]
```diff
@@ -36,4 +39,6 @@
     SessionPropertiesOverrideStrategy getSessionPropertiesOverrideStrategy();
 
     Map<String, String> getSessionPropertiesOverride();
+
+    Set<String> getSessionPropertiesRemoval();
```
Comment (mayankgarg1990) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfigurationOverrides.java:43]: `getSessionPropertiesToRemove()`

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfigurationOverridesConfig.java [hunk 3]
```diff
@@ -36,6 +38,7 @@
     private Optional<String> passwordOverride = Optional.empty();
     private SessionPropertiesOverrideStrategy sessionPropertiesOverrideStrategy = NO_ACTION;
     private Map<String, String> sessionPropertiesOverride = ImmutableMap.of();
+    private Set<String> sessionPropertiesRemoval = ImmutableSet.of();
```
Comment (mayankgarg1990) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfigurationOverridesConfig.java:41]: `sessionPropertiesToRemove`

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfigurationOverridesConfig.java [hunk 4]
```diff
@@ -107,6 +110,28 @@ public QueryConfigurationOverridesConfig setSessionPropertiesOverrideStrategy(Se
...
+        if (sessionPropertyRemoval == null) {
+            return this;
+        }
+
+        try {
+            this.sessionPropertiesRemoval = new ObjectMapper().readValue(sessionPropertyRemoval, new TypeReference<Set<String>>() {});
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
```
Comment (mayankgarg1990) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfigurationOverridesConfig.java:131]: should we just say something like `;` separated values ? That way specifying it is easier and the logic here is simpler as well

**Review by mayankgarg1990 — CHANGES_REQUESTED** [review]

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfiguration.java (hunk 1 above)
Comment (basarhamdionat) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfiguration.java]: Removed the settings.

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfigurationOverridesConfig.java (hunk 3 above)
Comment (basarhamdionat) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfigurationOverridesConfig.java]: ✅

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfigurationOverrides.java (hunk 2 above)
Comment (basarhamdionat) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfigurationOverrides.java]: ✅

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfigurationOverridesConfig.java (hunk 4 above)
Comment (basarhamdionat) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfigurationOverridesConfig.java]: I prefer comma separated as well, and  I can use Splitter and do that. 
But for Map above we are using Json Serialized format. Is that a standard for PrestoDB?

Comment (tdcmeehan) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfigurationOverridesConfig.java]: For JDBC URI parameters we separate key and value with `=` and different key-value pairs are separated by `;`.

Comment (basarhamdionat) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfigurationOverridesConfig.java]: ✅

File: presto-verifier/src/test/java/com/facebook/presto/verifier/framework/TestQueryConfigurationOverridesConfig.java [hunk 5]
```diff
@@ -48,14 +49,16 @@ public void testExplicitPropertyMappings()
                 .put("password-override", "_password")
                 .put("session-properties-override-strategy", "SUBSTITUTE")
                 .put("session-properties-override", "{\"key\": \"value\"}")
+                .put("session-properties-removal", "key2")
```
Comment (mayankgarg1990) [inline presto-verifier/src/test/java/com/facebook/presto/verifier/framework/TestQueryConfigurationOverridesConfig.java:52]: nit - pass in multiple values to test the splitting logic - `key3,key4`

**Review by mayankgarg1990 — APPROVED** [review]

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfiguration.java [hunk 6]
```diff
@@ -64,6 +64,10 @@ else if (overrides.getSessionPropertiesOverrideStrategy() == SUBSTITUTE) {
...
         }
 
+        for (String sessionPropertyToRemove : overrides.getSessionPropertiesToRemove()) {
+            sessionProperties.remove(sessionPropertyToRemove);
+        }
```
[the remark this answers is not in the document]
Comment (basarhamdionat) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfiguration.java]: ✅

File: presto-verifier/src/test/java/com/facebook/presto/verifier/framework/TestQueryConfigurationOverridesConfig.java (hunk 5 above)
Comment (basarhamdionat) [reply presto-verifier/src/test/java/com/facebook/presto/verifier/framework/TestQueryConfigurationOverridesConfig.java]: ✅

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**